sdk/src/azure/core/az_http_request.c

```diff
@@ -97,6 +101,66 @@ AZ_NODISCARD az_result az_http_request_append_path(_az_http_request* ref_request
   return AZ_OK;
 }
 
+// returns the query parameter value from a query paramenter name.
+// Or returns AZ_SPAN_NULL if query parameter name is not found
+static AZ_NODISCARD az_span
+_az_http_request_find_query_parameter(_az_http_request* ref_request, az_span query_paramenter_name)
+{
+  if (ref_request->_internal.query_start == 0)
+  {
+    return AZ_SPAN_NULL;
+  }
+
+  az_span url = ref_request->_internal.url;
+  uint8_t* query_params_ptr = az_span_ptr(url);
+  uint8_t* new_qp_name_ptr = az_span_ptr(query_paramenter_name);
+  int32_t new_qp_name_size = az_span_size(query_paramenter_name);
+
+  // will be set to the start of the value of a qp equal to the new qp
+  int32_t value_index_start = -1;
+
+  for (int32_t is_query_start = 1, index = ref_request->_internal.query_start; // +1 to jump `?`
+       index < ref_request->_internal.url_length - new_qp_name_size;
+       index++)
+  {
+    if (query_params_ptr[index] == '&')
+    {
+      is_query_start = 1; // set next index to be queryStart
+      if (value_index_start > 0)
+      { // if this was set before, it means we found the end of the value we want to return
+        return az_span_slice(url, value_index_start, index);
+      }
+      continue;
+    }
+    if (is_query_start == 1 && query_params_ptr[index] == new_qp_name_ptr[0])
+    { // at this point, a qp name start with the same letter as the new qp
+      // Check if they have same size before comparing content by cheking if
+      // adding the size of the new qp we get to the `=` delimiter
+      if (query_params_ptr[index + new_qp_name_size] == '=')
+      { // at this point, we know size is the same, now check the contents
+        az_span existing_pq = az_span_slice(url, index, index + new_qp_name_size);
+        if (az_span_is_content_equal(existing_pq, query_paramenter_name))
+        {
+          // advance index to value start
+          index = index + new_qp_name_size + 1; // +1 to jump '='
+          // set qp value start
+          value_index_start = index;
+        }
+      }
+    }
+    is_query_start = 0;
+  }
+
+  if (value_index_start > 0)
+  { // getting here means we found the qp at the last position of url, so the value goes to the end
+    // of url
+    return az_span_slice(url, value_index_start, ref_request->_internal.url_length);
```

Comment:
nit:
```suggestion
    return az_span_slice_to_end(url, value_index_start);
```

Reply:
yeah, that was my first approach but it is actually wrong..  the `url_length` can be smaller than the url span size.
Imaging az_span url of size 100 with an url like `http://r.com?q=value`,  url_leght is 20 while az_span url size is 100, 
So az_span_slice_to_end will get `value` with size greater than 5 which is not correct